specific/cmo_stock/models/stock.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        default=lambda self: self._get_partner_id(),
+    )
+    default_operating_unit_id = fields.Many2one(
+        'operating.unit',
+        string='Default Operating Unit',
+        compute='_compute_default_operating_unit_id',
+    )
+
+    @api.model
+    def _get_partner_id(self):
+        user = self.env['res.users'].browse(self._uid)
+        return user.partner_id and user.partner_id.id or False
+
+    @api.multi
+    @api.depends('partner_id')
+    def _compute_default_operating_unit_id(self):
+        for picking in self:
+            User = picking.env['res.users']
+            picking.default_operating_unit_id = False
+            if picking.partner_id:
+                user = User.search([('partner_id', '=',
+                                     picking.partner_id.id)])
+                if user:
+                    picking.default_operating_unit_id = \
+                        user[0].default_operating_unit_id.id
+
+    @api.multi
+    def action_open_stock(self):
+        # Get product_id in each line
+        product_ids = []
+        for picking in self:
+            product_ids += picking.move_lines.mapped('product_id.id')
+        return {
+            'name': _('Cuurent Inventory Valuation'),
+            'res_model': 'stock.history',
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'tree,graph',
+            'domain': "[('product_id','in', \
+                [" + ','.join(map(str, product_ids)) + "])]",
+            'context': "{'search_default_group_by_location': True, \
+                         'search_default_group_by_product': True}"
+        }
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    project_id = fields.Many2one(
+        'project.project',
+        string='Project name',
+        domain=lambda self: self._get_domain(),
+    )
+    location_dest_id = fields.Many2one(
+        'stock.location',
+        default=lambda self: self._get_location_dest_id(),
+    )
+
+    @api.model
+    def _get_domain(self):
+        user = self.env['res.users'].browse(self._uid)
+        operating_unit_ids = []
+        for operating_unit in user.operating_unit_ids:
+            operating_unit_ids.append(operating_unit.id)
+        return [('operating_unit_id', 'in', operating_unit_ids)]
+
+    @api.model
+    def _get_location_dest_id(self):
+        Location = self.env['stock.location']
+        user = self.env['res.users'].browse(self._uid)
```

Comment:
@newtratip , please help me review all the code about default and domain in the field in many places. I think we can cut it much shorter.